File: 4 - listas/ListasComObjetos/java/Aluno.java

public class Aluno {
	String nome;
	int idade;
	String curso;
	
	public Aluno(String nome, int idade, String curso) {
		this.nome = nome;
		this.idade = idade;
		this.curso = curso;
	}
	
	public String getNome() {
		return nome;
	}
	
	public void setNome(String nome) {
		this.nome = nome;
	}
	
	public int getIdade() {
		return idade;
	}
	
	public void setIdade(int idade) {
		this.idade = idade;
	}
	
	public String getCurso() {
		return curso;
	}
	
	public void setCurso(String curso) {
		this.curso = curso;
	}

    @Override
    public String toString() {
        return "[nome=" + nome + ", idade=" + idade + ", curso=" + curso + "]";
    }  

    @Override
    public boolean equals(Object obj) {
        Aluno other = (Aluno) obj;
        if (!nome.equals(other.nome))
            return false;
        if (idade != other.idade)
            return false;
        return true;
    }

    
}

File: 4 - listas/ListasComObjetos/c#/Aluno.cs

public class Aluno 
{
	public string Nome {get;set;}
	public int Idade {get;set;}
	public string Curso {get;set;}
	
	public Aluno(string nome, int idade, string curso) 
    {
		this.Nome = nome;
		this.Idade = idade;
		this.Curso = curso;
	}
	
    public override string ToString() {
        return "[nome=" + Nome + ", idade=" + Idade + ", curso=" + Curso + "]";
    }  
    
    public override bool Equals(Object obj) {
        Aluno other = (Aluno) obj;
        if (!Nome.Equals(other.Nome))
            return false;
        if (Idade != other.Idade)
            return false;
        return true;
    }

    
}
